src/java.base/share/classes/java/lang/foreign/SegmentAllocator.java

```diff
@@ -153,6 +153,54 @@ default MemorySegment allocateFrom(String str, Charset charset) {
         return segment;
     }
 
+    /**
+     * Converts a Java string into a null-terminated C string using the provided charset,
+     * and storing the result into a memory segment.
+     * <p>
+     * This method always replaces malformed-input and unmappable-character
+     * sequences with this charset's default replacement byte array. The
+     * {@link java.nio.charset.CharsetEncoder} class should be used when more
+     * control over the encoding process is required.
+     * <p>
+     * If the given string contains any {@code '\0'} characters, they will be
+     * copied as well. This means that, depending on the method used to read
+     * the string, such as {@link MemorySegment#getString(long)}, the string
+     * will appear truncated when read again.
+     *
+     * @param str      the Java string to be converted into a C string
+     * @param charset  the charset used to {@linkplain Charset#newEncoder() encode} the
+     *                 string bytes
+     * @param srcIndex the starting index of the source string
+     * @param numChars the number of characters to be copied
+     * @return a new native segment containing the converted C string
+     * @throws IllegalArgumentException if {@code charset} is not a
+     *         {@linkplain StandardCharsets standard charset}
```

Comment:
Same here, I don't think we have this limitation.

Reply:
Done